mlir-clang/Lib/clang-mlir.h

```diff
@@ -46,6 +46,9 @@ struct LoopContext {
   mlir::Value noBreak;
 };
 
+// TODO: 1. split this in new file 'ValueWithOffset.h'?
```

Comment:
@wsmoses Hi Billy, I would like to do the following for ValueWithOffset class, but before going down this rabbit hole, let's agree:
1. Move the class to a new file ValueWithOffset.h
2. Make the struct a class and make 'val' and 'isReference' private, add additional public methods 'get' similar to what we have for the AffineLoopDescriptor
3. Add doc
4. Move methods in a .cpp file
5. I would like to rename 'getValue' to something like 'loadVariable'. 'getValue' does not say much
6. Same for 'store' I would like to rename to 'storeVariable'